### PR TITLE
DOC: Updates on popper workflow and .popper.yml structure

### PR DESCRIPTION
**Pull request by UjjwalAyyangar**
Part of #355 
* Adds details about popper workflow
* Updates explanation on .popper.yml configuration file

### CODE REVIEW

File: docs/sections/cli_features.md [hunk 1]
```diff
@@ -122,7 +122,26 @@ file of the pipeline.
 
 ## Visualizing a pipeline
 
-**TODO**
+Popper gives a user the ability to visualize the workflow of a pipeline using the
+`popper workflow pipeline_name` command. The command generates a workflow diagram 
+corresponding to a Popper pipeline, in the .dot format. The string defining
+the graph is printed to stdout so it can be piped into other tools. 
+For example,to generate a png file, one can make use of the graphviz CLI tools: 
+popper workflow mypipe | dot -T png -o mypipe.png.
```
Comment (ivotron) [inline docs/sections/cli_features.md]: add this as a code snippet:

```bash
popper workflow mypipe | dot -T png -o mypipe.png
```

File: docs/sections/cli_features.md [hunk 2]
```diff
@@ -122,7 +122,26 @@ file of the pipeline.
...
 
-**TODO**
+Popper gives a user the ability to visualize the workflow of a pipeline using the
+`popper workflow pipeline_name` command. The command generates a workflow diagram 
+corresponding to a Popper pipeline, in the .dot format. The string defining
+the graph is printed to stdout so it can be piped into other tools. 
+For example,to generate a png file, one can make use of the graphviz CLI tools: 
+popper workflow mypipe | dot -T png -o mypipe.png. 
+
+Example usage:
```
Comment (ivotron) [inline docs/sections/cli_features.md]: Remove this line

File: docs/sections/cli_features.md [hunk 3]
```diff
@@ -122,7 +122,26 @@ file of the pipeline.
...
+
+Suppose you want to visualize the [co2-emissions](https://github.com/popperized/swc-lesson-pipelines/tree/master/pipelines/co2-emissions) pipeline. 
+Assuming that this pipeline is added to your repository ( as explained in `Searching and Importing pipelines` ), you need to type:
```
Comment (ivotron) [inline docs/sections/cli_features.md]: * remove spaces after open/close parenthesis
  * add a link to that section

File: docs/sections/cli_features.md [hunk 4]
```diff
@@ -214,20 +234,20 @@ init paper` and has by default a single stage named `build.sh`.
 
 #### `envs`
 
-The `envs` entry in `.popper.yml` specifies the environment that a 
-pipeline is used when the pipeline is executed as part of the `popper 
+The `envs` entry in `.popper.yml` specifies the environment in which a 
+pipeline is used, when the pipeline is executed as part of the `popper
```
Comment (ivotron) [inline docs/sections/cli_features.md]: "in which a pipeline is executed as part of the `popper run` command."

File: docs/sections/cli_features.md [hunk 5]
```diff
@@ -272,26 +292,45 @@ pipeline showed in the example.
...
 ### Metadata
 
-The `metadata` YAML entry specifies the set of data that gives information
-about the user's project. It can be added using the `popper metadata --add command`
-For example :
+The `metadata` YAML entry specifies a set of key-value pairs that
+describes and gives us information about a project.
+
+By default, a project's metadata will be initialized with the
+following key-value pairs :-
```
Comment (ivotron) [inline docs/sections/cli_features.md]: replace `:-` with `:`

**Comment by UjjwalAyyangar** [comment]
@ivotron Requested changes have been made.

**Comment by ivotron** [comment]
thanks a lot!

**Pull request merged**